apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js

```diff
@@ -162,28 +162,67 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
+                const existingServiceConsumers = await ServiceConsumer.getAll(context, {
+                    OR: serviceConsumersData.map((data) => ({
+                        AND: [
+                            { resident: { id: data.resident } },
+                            { accountNumber: data.accountNumber },
+                            { organization: { id: data.organization } },
+                        ],
+                    })),
+                })
+
                 const createdServiceConsumers = await Promise.all(
-                    definedCombinations.map(([resident, account]) => {
-                        const organizationId = get(account, ['context', 'organization', 'id'], null)
-                        const billingContextId = get(account, ['context', 'id'], null)
-                        const [acquiringContextId] = get(organizationsToAcquiringContextsMap, organizationId, [null])
-
-                        return ServiceConsumer.create(context, {
-                            dv,
-                            sender,
-                            resident: { connect: { id: resident.id } },
-                            accountNumber: account.number,
-                            organization: { connect: { id: organizationId } },
-                            billingAccount: { connect: { id: account.id } },
-                            billingIntegrationContext: billingContextId ? { connect: { id: billingContextId } } : null,
-                            acquiringIntegrationContext: acquiringContextId ? { connect: { id: acquiringContextId } } : null,
-                            isDiscovered: true,
+                    serviceConsumersData.map((serviceConsumerData) => {
+                        const [existingServiceConsumer] = filter(existingServiceConsumers, {
+                            resident: { id: serviceConsumerData.resident },
+                            accountNumber: serviceConsumerData.accountNumber,
+                            organization: { id: serviceConsumerData.organization },
                         })
+
+                        const data = {
+                            ...serviceConsumerData,
+                            resident: { connect: { id: serviceConsumerData.resident } },
+                            organization: { connect: { id: serviceConsumerData.organization } },
+                            billingAccount: { connect: { id: serviceConsumerData.billingAccount } },
+                            billingIntegrationContext: serviceConsumerData.billingIntegrationContext ? { connect: { id: serviceConsumerData.billingIntegrationContext } } : null,
+                            acquiringIntegrationContext: serviceConsumerData.acquiringIntegrationContext ? { connect: { id: serviceConsumerData.acquiringIntegrationContext } } : null,
+                        }
+
+                        if (existingServiceConsumer) {
```

Comment:
Do we need this update?

Reply:
Why not? We have the same logic at manual creating of service consumer.
Looks like the main purpose is to restore deleted model.